File: solutions/2026-07-15/solution.py
# Reverse Linked List
# Difficulty: Easy
# URL: https://leetcode.com/problems/reverse-linked-list/

class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

def reverseList(self, head: ListNode) -> ListNode:
    if not head or not head.next:
        return head

    new_head = head
    while head.next:
        next_node = head.next
        head.next = next_node.next
        next_node.next = new_head
        new_head = next_node
    return new_head

File: solutions/2026-07-15/solution.java
// Reverse Linked List
// Difficulty: Easy
// URL: https://leetcode.com/problems/reverse-linked-list/

class ListNode {
    int val;
    ListNode next;

    ListNode(int x) {
        val = x;
        next = null;
    }

    public ListNode reverseList(ListNode head) {
        if (head == null || head.next == null) {
            return head;
        }

        ListNode newHead = head;
        while (head.next != null) {
            ListNode nextNode = head.next;
            head.next = nextNode.next;
            nextNode.next = newHead;
            newHead = nextNode;
        }
        return newHead;
    }
}